### PR TITLE
6 decision tree introspection

### PR DESCRIPTION
**Pull request by Coop8**
Introspection done and tested! Yippie!

### CODE REVIEW

File: test/Sandbox.py
Comment (galaxybomb23) [inline test/Sandbox.py:1]: don't remove these tests

**Review by galaxybomb23 — CHANGES_REQUESTED** [review]

File: src/DTIntrospection.cpp [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+
+            for (int i = 0; i < featuresAndValues.size(); i++) {
+                featuresAndValues[i].insert(0, featureNames[i] + "=");
+            }
+            
+            for (const auto& item : featuresAndValues) {
+                FeatureOpValue featureOpValue2 = extractFeatureOpValue(item);
+                double value2AsDouble = convert(featureOpValue2.value);
+                
+                if (!std::isnan(valueAsDouble) && !std::isnan(value2AsDouble)) {
```
Comment (iBoot32) [inline src/DTIntrospection.cpp:599]: Logic is duplicated, would be a good candidate for using a lambda or something in a templated function.

File: src/DTIntrospection.cpp [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+FeatureOpValue DTIntrospection::extractFeatureOpValue(string featureValueCombo) {
+    std::regex pattern1(R"((.+)=(.+))");
+    std::regex pattern2(R"((.+)<(.+))");
+    std::regex pattern3(R"((.+)>(.+))");
+
+    string feature, op, value;
+    std::smatch match;
+
+
+    if (std::regex_search(featureValueCombo, match, pattern1)) {
```
Comment (iBoot32) [inline src/DTIntrospection.cpp:641]: We discussed this a bit, probably faster to not use regex but not a huge deal.

File: test/Sandbox.py
Comment (Coop8) [reply test/Sandbox.py:1]: Fixed

File: src/DTIntrospection.cpp (hunk 2 above)
Comment (Coop8) [reply src/DTIntrospection.cpp:641]: Yes, could be an optimization in the future, but for now, it works.

File: src/DTIntrospection.cpp (hunk 1 above)
Comment (Coop8) [reply src/DTIntrospection.cpp:599]: Could be another optimization in the future, noted.

**Review by galaxybomb23 — APPROVED** [review]

**Pull request merged**